src/irs/core/s3.py

```diff
@@ -15,9 +15,9 @@
 """Provides helpers for S3 interaction"""
 
 import math
-from abc import ABC, abstractmethod
 from collections.abc import AsyncGenerator, Sequence
 from dataclasses import dataclass
+from typing import Optional
```

Comment:
We have so far avoided terms in the domain core that are directly hinting at the implementation technology (AWS S3) of the object storage. Thus I would propose to rename the module to "staging_handler".